Scripts/Runtime/Entities/Lifecycle/PrototypeVariant.cs

```diff
@@ -0,0 +1,75 @@
+using Anvil.Unity.DOTS.Core;
+using Anvil.Unity.DOTS.Util;
+using System;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    public readonly struct PrototypeVariant : IEquatable<PrototypeVariant>,
+                                              IComponentData,
+                                              IToFixedString<FixedString32Bytes>
+    {
+        public static readonly int UNSET_VALUE = default;
```

Comment:
Yeah I had wanted that to, but until we get [C# 10's parameterless struct constructors ](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/parameterless-struct-constructors), I feel like the current implementation is the safest. 

I can't prevent someone from creating a new `PrototypeVariant` because just declaring it creates the `default` instance. 

If someone uses that I can detect that they are using the `UNSET_VALUE` and complain.

If someone explicitly calls the constructor and passes the `UNSET_VALUE` I can also detect that and trigger the assert. 

The downside is that you have to be explicit in your `enum` and not use 0 but that seems like a low amount of friction for the safety.

If I make the `UNSET_VALUE` be `-1` or `int.MinValue` there's no way it ever gets set to that unless you as the developer explicitly pass it in.